mars_street_view/models.py

```diff
@@ -1,27 +1,205 @@
+"""Defines models for the whole project: Photo, Rover and Camera."""
+from __future__ import unicode_literals, print_function
 from sqlalchemy import (
     Column,
-    Index,
     Integer,
-    Text,
-    )
+    String,
+    ForeignKey,
+    Text
+)
 
 from sqlalchemy.ext.declarative import declarative_base
 
 from sqlalchemy.orm import (
     scoped_session,
     sessionmaker,
-    )
+    relationship
+)
+
+from sqlalchemy.orm.exc import (
+    MultipleResultsFound,
+    NoResultFound,
+)
 
 from zope.sqlalchemy import ZopeTransactionExtension
 
+from sqlalchemy import func
+
 DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
 Base = declarative_base()
 
+LEFT_LENS_URL = '%L___-BR.JPG'
+RIGHT_LENS_URL = '%R___-BR.JPG'
+LOW_RES_SPI_OPP = '%ESF_____________-BR.JPG'
+BAD_CUR_1 = '%_M_______NCAM_______.JPG'
+BAD_CUR_2 = '%SAPP_______.JPG'
+BAD_CUR_3 = '%_D_______TRAV_______.JPG'
+
 
 class MyModel(Base):
+    """Test model."""
+
     __tablename__ = 'models'
     id = Column(Integer, primary_key=True)
     name = Column(Text)
     value = Column(Integer)
 
-Index('my_index', MyModel.name, unique=True, mysql_length=255)
+
+class Photo(Base):
+    """Each individual photo object from a NASA API query."""
+
+    def __init__(self, id=None, camera=None, rover=None, **kwargs):
+        """Initialize the Photo object, renaming some parameters."""
+        kwargs['nasa_id'] = id
+        try:
+            rover_name = rover['name']
+            kwargs['rover_name'] = rover_name
+        except AttributeError:
+            raise KeyError('Photo must be initialized with a rover obj.')
```

Comment:
Also, it should be that if you set photo.rover to a rover instance, you'll get rover_name for free, since that is the field on which the relationship is built.  Right?